Core/GDCore/Events/Parsers/ExpressionParser2.h

```diff
@@ -547,28 +547,28 @@ class GD_CORE_API ExpressionParser2 {
   }
   ///@}
 
-  std::unique_ptr<ExpressionParserDiagnostic> ValidateOperator(
+  std::unique_ptr<ExpressionParserError> ValidateOperator(
       gd::String::value_type operatorChar) {
     if (operatorChar == '+' || operatorChar == '-' || operatorChar == '/' ||
         operatorChar == '*') {
-      return gd::make_unique<ExpressionParserDiagnostic>();
+      return std::unique_ptr<ExpressionParserError>(nullptr);

```

Comment:
This change is not just a refactoring. It was wrongly replacing the null pointer with a default constructed instance.
When removing the base class with `IsError() { return false }`, it returned `true` which causes bugs.
`IsError()` no longer exist, only null is checked.

Reply:
Thanks for the fix 👍 Much better